Spectral Normalization

yseongjin opened the pull request:
I added spectral normalization 'SpectralNorm.py' to util directory.

[Usage]
from SpectralNorm import spectral_norm

def discriminator_block(in_filters, out_filters, bn=True):
            block = [   spectral_norm(nn.Conv2d(in_filters, out_filters, 3, 2, 1)),
                        nn.LeakyReLU(0.2, inplace=True),
                        nn.Dropout2d(0.25)]
            if bn:
                block.append(nn.BatchNorm2d(out_filters, 0.8))
            return block

yseongjin committed:
Spectral Nomalization

/util/SpectralNorm.py

yseongjin committed:
Update Spectral Normalization

Update doctstring

yseongjin committed:
Delete temp file

Delete .un~ file in util

whikwon commented:
TrainingTest.ipynb는 혹시 빼주실 수 있을까요.

테스트 하신 내용을 스크립트로 만들어서 test_spectralnorm.py 로 작성해서 올려주시면 더 좋을 듯 합니다.!
혹시, 위 스크립트 가져온 내용이면 링크를 알 수 있을까요?

yseongjin committed:
Test Example of Spectral Normalization

test_spectral.py

yseongjin committed:
Delete TrainingTest.ipynb

yseongjin committed:
Delete Test Code of Spectral Normalization.

Delete test_spectral.py

yseongjin commented:
1. TrainingTest.ipynb 삭제
2. Slack에 test_spectral.py 전달
완료 했습니다.

whikwon commented:
test_spectral.py pytest로 통과하는지 확인하고 함께 commit 해도 좋을거 같습니다.!

yseongjin committed:
Apply unpack operator in Spectral Nomalization

In compute_weight method, apply * operator to tensor permute
method call.

yseongjin committed:
Appy * operator to permute function call

In compute_wieght method of SpectralNorm class,
appy * operator to other_axes list for passing unpacked
dimension list to permute method call.
